lib/cli_script.dart

```diff
@@ -31,6 +34,54 @@ export 'src/parse_args.dart' show arg;
 export 'src/script.dart';
 export 'src/stdio.dart' hide stdoutKey, stderrKey;
 
+/// The packages whose stack frames should be folded by [wrapMain].
+const _packagesToFold = {'cli_script', 'async', 'path', 'string_scanner'};
+
+/// A wrapper for the body of the top-level `main()` method.
+///
+/// It's recommended that all scripts wrap their `main()` methods in this
+/// function. It gracefully exits when an error is unhandled, and (if
+/// [chainStackTraces] is `true`) tracks stack traces across asynchronous calls
+/// to produce better error stacks.
+///
+/// By default, if a [Script] fails in [callback], this just exits with that
+/// script's exit code without printing any additional error messages. If
+/// [printScriptException] is `true`, this will print the exception and its
+/// stack trace to stderr before exiting.
+///
+/// By default, stack frames from the Dart core libraries and from this package
+/// will be folded together to make stack traces more readable. If
+/// [verboseTrace] is `true`, the full stack traces will be printed instead.
+void wrapMain(FutureOr<void> callback(),
+    {bool chainStackTraces = true,
+    bool printScriptException = false,
+    bool verboseTrace = false}) {
+  Chain.capture(callback, onError: (error, chain) {
+    if (!verboseTrace) {
+      chain = chain.foldFrames(
+          (frame) => _packagesToFold.contains(frame.package),
+          terse: true);
+    }
+
+    if (error is! ScriptException) {
+      stderr.writeln(error);
+      stderr.writeln(chain);
+      exit(254);
```

Comment:
Is there a specific meaning behind this exit code?

Reply:
It's the exit code Dart uses for unhandled exceptions. I'll add a comment.